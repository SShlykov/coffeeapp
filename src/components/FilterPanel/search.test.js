import React from 'react';
import Search from "./Seach";
import {shallow} from 'enzyme';

describe('Testing <Search/>', () => {
    const seach = shallow(<Search id="1" type="best"/>);
    describe('Testing snap & state', () => {
        it('Search bar have rendered correctly', () => {
            expect(seach).toMatchSnapshot();
        });
        it('Search term state is exist', () => {
            expect(seach.state().term).toBe("");
        });        
    });
});