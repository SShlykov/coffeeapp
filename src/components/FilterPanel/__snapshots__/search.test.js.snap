// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Search/> Testing snap & state Search bar have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    id="1"
    type="best"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <label
          className="shop__search-label"
          htmlFor="filter"
        >
          Looking for
        </label>,
        <input
          className="shop__search-input"
          id="filter"
          onChange={[Function]}
          placeholder="start typing here..."
          type="text"
          value=""
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Looking for",
          "className": "shop__search-label",
          "htmlFor": "filter",
        },
        "ref": null,
        "rendered": "Looking for",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "shop__search-input",
          "id": "filter",
          "onChange": [Function],
          "placeholder": "start typing here...",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <label
            className="shop__search-label"
            htmlFor="filter"
          >
            Looking for
          </label>,
          <input
            className="shop__search-input"
            id="filter"
            onChange={[Function]}
            placeholder="start typing here..."
            type="text"
            value=""
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Looking for",
            "className": "shop__search-label",
            "htmlFor": "filter",
          },
          "ref": null,
          "rendered": "Looking for",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "shop__search-input",
            "id": "filter",
            "onChange": [Function],
            "placeholder": "start typing here...",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
